### ISSUE TITLE
Button Spacing is too specific.

### ISSUE THREAD

**Issue by thezacharytaylor** [a11yphant/a11yphant#188]
If you think you've found a bug or accessibility issue, feel free to let us know. Before you continue filling out this issue, please answer this checklist for yourself:

1. I ensured that the problem is not caused by my local environment (e.g., a browser plugin, a really old browser version or unusual browser settings)
2. I checked that no GitHub issue was already filed for this bug

If you answered everything with yes, go ahead with this GitHub issue. Be sure to include as much information as possible so we can reproduce the bug.

**Describe the bug**
The button spacing challenge needs `margin: 10px 10px;`. Failing to add this specifically fails the user, even if you get it right.

**To Reproduce**
On Button Challenge 07:
If you add `margin-top: 10px` and `margin-bottom: 10px;` to the button element and then make individual rules for margin-right and left for the first and last of type (10px each) it fails. But if you replace just `margin-top: 10px` and `margin-bottom: 10px;` with `margin: 10px 0;` it will pass. You can even leave those two rules and have them preceded them with `margin: 10px 0;` and it will work, but not if you lower them below 10px despite keeping `margin` at 10px (which is correct behavior).

Again, the problem is: `margin: ...` needs to be present and greater than 10px despite the -top and -bottom either doing the same job or even overriding it with the correct values. Say this satisfies the task, but fails the check:

```css
/* Example. */
button {
	min-height: 44px;
	min-width: 44px;
        margin: 5px 0;
	margin-top: 10px;
	margin-bottom: 10px;
}

button:first-of-type {
	margin-right: 10px;
}

button:last-of-type {
	margin-left: 10px;
}
```
The inspector shows the 10px top and bottom margin despite the failure.

My original 'failing' code below
```css
/* This, while correct, fails. */
button {
	min-height: 44px;
	min-width: 44px;
        /* Because the following is too specific. */
	margin-top: 10px;
	margin-bottom: 10px;
}

button:first-of-type {
	margin-right: 10px;
}

button:last-of-type {
	margin-left: 10px;
}
```

**Expected behaviour**
To pass.

**Screenshots**
Probably not necessary.

**Desktop (please complete the following information):**
 - OS: MacOS Ventura 13.6.3
 - Arc Version 1.21.3 

**Additional context**
None.

**Comment by lucapircher** [comment]
Hi @thezacharytaylor,
thanks for reporting this!

Unfortunately determining the spacing between two elements without actually rendering them in a browser is quite tricky. As you are certainly aware CSS provides lots possibilities. But I definitely see room for improving this check by taking into account that multiple margin declarations might be present.

**Comment by thomasdax98** [comment]
Hi @thezacharytaylor,
Thanks again for reporting this. We improved the evaluation. Now we also take `margin-left` and `margin-right` into account during evaluation, so your solution works. 

We are aware that there are other ways than margin to solve this challenge. The best evaluation would be to calculate the rendered space between the two elements. But we can’t do that because we don’t actually render the submitted code during evaluation (anymore).
In the past, we used Selenium for rendering. However, we abandoned that approach last year to reduce cost and simplify our setup (since we are doing a11yphant purely as a hobby). Now we use JSDOM instead and this is one of its downsides.
